cmd/main.go

```diff
@@ -434,7 +436,35 @@ func autoMigrationWithLeader(migrationLeader leader.ElectorInterface, db *gorm.D
 }
 
 func uploadBaseISOWithLeader(uploadLeader leader.ElectorInterface, objectHandler s3wrapper.API, generator generator.ISOInstallConfigGenerator, log logrus.FieldLogger) error {
```

Comment:
maybe this section should move to a different package? it's a lot of code.

Reply:
@filanov I agree with you, but I am already almost done with a PR that is based on this, and I am moving the code to s3wrapper together with code from my PR that uploads PXE artifacts.  Suggest we continue with this and my PR will move it.

Reply:
No problem but then you will need to write tests for everything :D

Reply:
Will add tests for this, sure